lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php

```diff
@@ -45,6 +52,14 @@ class DB2Statement implements IteratorAggregate, Statement
     /** @var mixed[] */
     private $bindParam = [];
 
+    /**
+     * Map of LOB parameter positions to the tuples containing reference to the variable bound to the driver statement
+     * and the temporary file handle bound to the underlying statement
```

Comment:
@morozov are the file handles cleaned up at `__destruct()`? AFAIK, this stuff is a potential memleak.

Reply:
> The file is automatically removed when closed (for example, by calling `fclose()`, or when there are no remaining references to the file handle returned by `tmpfile()`), or when the script ends.

See `::execute()`.

Reply:
I see:

```php
        // release temporary file handles
        $this->lobs = [];
```

This does not guarantee that the resource handles were closed: sadly, GC does not work the same way for objects and resources.

Reply:
We can close them explicitly.

Reply:
That's what I was getting at: `fclose()` is kinda required here

Reply:
`tmpfile()` are treated specially, you don't need to manually GC them, they'll be destroyed once there are no more references. See [documentation](http://php.net/manual/en/function.tmpfile.php):
> The file is automatically removed when closed (for example, by calling fclose(), or when there are no remaining references to the file handle returned by tmpfile()), or when the script ends.

Reply:
Hmm, interesting quirk: resources are one of those nightmares that keep on giving :+1: